src/rln_contract.ts

```diff
@@ -0,0 +1,91 @@
+import { ethers } from "ethers";
+
+import { RLN_ABI } from "./const.js";
+import { RLNInstance } from "./rln.js";
+
+type Member = {
+  pubkey: string;
+  index: number;
+};
+
+export class RLNContract {
+  private contract: ethers.Contract;
+  private membersFilter: ethers.EventFilter;
+
+  private members: Member[] = [];
+
+  constructor(
+    address: string,
+    provider: ethers.Signer | ethers.providers.Provider
+  ) {
+    this.contract = new ethers.Contract(address, RLN_ABI, provider);
+    this.membersFilter = this.contract.filters.MemberRegistered();
+  }
+
+  public getContract(): ethers.Contract {
+    return this.contract;
+  }
+
+  public getMembers(): Member[] {
```

Comment:
Not great as the user needs to know to fetch the members before calling this method.

I'd recommend to at least document the fact that `subscriveToMembers` and `fetchMembers` should be done first.

I'd also recommend to make it part of the API that there is several stage and returned `undefined | Member[]`:

- `undefined` returned -> the fetching/subscription was not done yet
- empty array: no members in contracts